stdlib/int16.mli

```diff
@@ -0,0 +1,171 @@
+
+(** 16-bit integers.
+
+    This module provides operations on the type [t] of signed 16-bit integers.
+    All arithmetic operations over [t] are taken modulo 2{^16}.
+
+    Internally, values of type [t] are represented as [int].
+
+    @since 5.4 *)
+
+type t
+(** The type of 16-bit integers. *)
+
+val zero : t
+(** The 16-bit integer 0. *)
+
+val one : t
+(** The 16-bit integer 1. *)
+
+val minus_one : t
+(** The 16-bit integer -1. *)
+
+val neg : t -> t
+(** Unary negation. *)
+
+val add : t -> t -> t
+(** Addition. *)
+
+val sub : t -> t -> t
+(** Subtraction. *)
+
+val mul : t -> t -> t
+(** Multiplication. *)
+
+val div : t -> t -> t
+(** Integer division. This division rounds the real quotient of its arguments
+    towards zero, as specified for {!Stdlib.(/)}. @raise Division_by_zero if the
+    second argument is zero. *)
+
+val unsigned_div : t -> t -> t
+(** Same as {!div}, except that arguments and result are interpreted as {e
+    unsigned} 16-bit integers. *)
+
+val rem : t -> t -> t
+(** Integer remainder. If [y] is not zero, the result of [Int16.rem x y]
+    satisfies the following property:
+    [x = Int16.add (Int16.mul (Int16.div x y) y) (Int16.rem x y)]. If [y = 0],
+    [Int16.rem x y] raises [Division_by_zero]. *)
+
+val unsigned_rem : t -> t -> t
+(** Same as {!rem}, except that arguments and result are interpreted as {e
+    unsigned} 16-bit integers. *)
+
+val succ : t -> t
+(** Successor. [Int16.succ x] is [Int16.add x Int16.one]. *)
+
+val pred : t -> t
+(** Predecessor. [Int16.pred x] is [Int16.sub x Int16.one]. *)
+
+val abs : t -> t
+(** [abs x] is the absolute value of [x]. On [min_int] this is [min_int] itself
+    and thus remains negative. *)
+
+val max_int : t
+(** The greatest representable 16-bit integer, 2{^15} - 1. *)
+
+val min_int : t
+(** The smallest representable 16-bit integer, -2{^15}. *)
+
+val logand : t -> t -> t
+(** Bitwise logical and. *)
+
+val logor : t -> t -> t
+(** Bitwise logical or. *)
+
+val logxor : t -> t -> t
+(** Bitwise logical exclusive or. *)
+
+val lognot : t -> t
+(** Bitwise logical negation. *)
+
+val shift_left : t -> int -> t
+(** [Int16.shift_left x y] shifts [x] to the left by [y] bits. The result is
+    unspecified if [y < 0] or [y >= 16]. *)
+
+val shift_right : t -> int -> t
+(** [Int16.shift_right x y] shifts [x] to the right by [y] bits. This is an
+    arithmetic shift: the sign bit of [x] is replicated and inserted in the
+    vacated bits. The result is unspecified if [y < 0] or [y >= 16]. *)
+
+val shift_right_logical : t -> int -> t
+(** [Int16.shift_right_logical x y] shifts [x] to the right by [y] bits. This is
+    a logical shift: zeroes are inserted in the vacated bits regardless of the
+    sign of [x]. The result is unspecified if [y < 0] or [y >= 16]. *)
+
+val of_int : int -> t
+(** Convert the given integer (type [int]) to a 16-bit integer (type [t]). The
+    argument is taken modulo 2{^16}. *)
```

Comment:
What is the behaviour here for values larger than 16-bit int? Same question for 8-bit. 

Note the docs on both Int64 and Int32 don't mention the behaviour either.

Reply:
Not to get caught up in that discussion too much.  The small amount of extra doc on Int64.to_int about loosing the top 32 bits feels useful to me. 
https://ocaml.org/manual/5.3/api/Int64.html

Reply:
> Note the docs on both Int64 and Int32 don't mention the behaviour either.